### ISSUE TITLE
Bringing `react-vector-graphics` up to date

### ISSUE THREAD

**Issue by mcataford** [tophat/react-vector-graphics#289]
This package needs to be brought back up _au goût du jour_ -- here is a living list of bits and bobs to resolve which can be tackled in chunks:

- [ ] Node is pinned to v10 by `.nvmrc` and should be looking at `lts/hydrogen` instead;
- [ ] Renovate is not pushing dependency updates to this repository;
- [ ] Dependencies are out of date;
- [ ] Yarn is not up-to-date and still expect Yarn v1;

**Comment by noahnu** [comment]
Taking a quick look to see if I can migrate it to yarn + monodeploy, copying commit-utils.

**Comment by mcataford** [comment]
Yeah, I was taking a brief look at that as a first step as well (mainly the Yarn part, which also has Node be updated). I'll need to read more on workspaces setup to understand why tests aren't passing out of the box even though.

If you have a direct path, have at it! :rocket:

**Comment by noahnu** [comment]
Update CI/typescript/linting/publish framework etc.. It's on manual publishing right now via workflow_dispatch. Just need to do some manual QA before we can publish a new version.

I had to ignore a number of TS errors to get things passing, which makes me lose some confidence in the system.

**Comment by mcataford** [comment]
Neato!

Coming back to this today.

I closed all of the old dependabot-related stale PRs and enabled Renovate for all repos (previously, it was manually turned on for select repos in the org). With the configuration you added, we should see the dashboard issue pop up soonish.

> Just need to do some manual QA before we can publish a new version.

Did you have a specific QA plan in mind?

**Comment by noahnu** [comment]
QA plan: test out the examples in the repo and also link to internal tooling to ensure there's no regressions, especially from the svgr update.

I took some shortcuts with the typescript types because I'm not familiar enough with the project to understand the intent of the types. There should be opportunity to simplify.

**Comment by noahnu** [comment]
Our internal tooling is in JS which makes it a bit harder to test. I think we'll want to convert the internal tooling to TypeScript and then work on fixing up the public types of react-vector-graphics as we do the migration. The types at the moment are conflating react-vector-graphics types with svgr types. So we're missing types on "options" for example, or the fact that we infer componentName.

We'll want to streamline this tool further (and our internal use of it) before releasing a breaking change. Alternatively, we migrate off of this and aim to deprecate this sooner.
